src/geometry/minkowski.md

```diff
@@ -0,0 +1,97 @@
+<!--?title Minkowski sum of convex polygons -->
+# Minkowski sum of convex polygons
+
+## Definition
+Consider two sets $A$ and $B$ of points on a plane. Minkowski sum $A + B$ is defined as $\\{a + b| a \in A, b \in B\\}$.
+Here we will consider the case when $A$ and $B$ consist of convex polygons $P$ and $Q$ with their interiors.
+It turns out that the sum of convex polygons $P$ and $Q$ is a convex polygon with at most $|P| + |Q|$ vertices.
+
+## Algorithm
+
+Here we consider the polygons to be cyclically enumerated, i. e. $P_{|P|} = P_0,\ Q_{|Q|} = Q_0$ and so on.
+
+Since the size of the sum is linear in terms of the sizes of initial polygons, we should aim at finding a linear-time algorithm.
+Suppose that both polygons are ordered counter-clockwise. Consider sequences of edges $\\{\overrightarrow{P_iP_{i+1}}\\}$
+and $\\{\overrightarrow{Q_jQ_{j+1}}\\}$ ordered by polar angle. We claim that the sequence of edges of $P + Q$ can be obtained by merging
+these two sequences preserving polar angle order and replacing consequitive co-directed vectors with their sum. Straightforward usage of this idea results
+in a linear-time algorithm, however, restoring the vertices of $P + Q$ from the sequence of sides requires repeated addition of vectors,
+which may introduce unwanted precision issues if we're working with floating-point coordinates, so we will describe a slight
+modification of this idea.
+
+
+Firstly we should reorder the vertices in such a way that the first vertex
+of each polygon has the lowest y-coordinate (in case of several such vertices pick the one with the smallest x-coordinate). After that the sides of both polygons
+will become sorted by polar angle, so there is no need to sort them manually.
+Now we create two pointers $i$ (pointing to a vertex of $P$) and $j$ (pointing to a vertex of $Q$), both initially set to 0.
+We repeat the following steps while $i < |P|$ or $j < |Q|$.
+
+1. Append $P_i + Q_j$ to $P + Q$.
+
+2. Compare polar angles of $\overrightarrow{P_iP_{i + 1}}$ and $\overrightarrow{Q_jQ_{j+1}}$.
+
+3. Increment the pointer which corresponds to the smallest angle (if the angles are equal, increment both).
+
+
+## Distance between two polygons
+One of the most common applications of Minkowski sum is computing the distance between two convex polygons (or simply checking whether they intersect).
+The distance between two convex polygons $P$ and $Q$ is defined as $\min\limits_{a \in P, b \in Q} ||a - b||$.
+If we reflect $Q$ through the point $(0, 0)$ obtaining polygon $-Q$, the problem boils down to finding a point in
+$P + (-Q)$ with the smallest distance to $(0, 0)$, which can be done in linear time. Since Minkowski sum can be computed
+in linear time, we obtain a linear-time algorithm for finding the distance between two convex polygons.
```

Comment:
Can you add a line, saying that the minimal distance between two polygons is not necessaryily the distance between two vertices, but can also be between a vertex and an edge. And similarly that the distance between the sum of the polygons and $(0, 0)$ is not necessary the distance between a vertex and (0, 0), but can also be the distance between an edge and $(0, 0)$.